Parsers/Alsid/afad_parser.kql

```diff
@@ -91,6 +91,16 @@ let Changes = Common
                      EventID:string "\" \""
                      EventType:string "\" "
                      Attributes:string "\r?\n";
-union Changes, Deviances
+let Attacks = Common
```

Comment:
There is no MessageType = 2 in the [Alsid sample data set](https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/Syslog/AlsidForADLog_CL.json), so I could not test that specifically.

Also, I think the overall parser isn't working for after you do the initial parse on RawData.  DistinctPart doesn't appear to have the strings you expect as the \" all get removed in my testing.  This means this subsequent parsing doesn't seem to work.  I changed the Deviances and below portions to the following and it seems to work properly:
```
let Deviances = Common
| where MessageType == 0 
| parse DistinctPart with '"'
                     Codename:string '" "'
                     Severity:string '" "'
                     ADObject:string '" "'
                     DevianceID:string '" "'
                     ProfileID:string '" "'
                     ReasonCodename:string '" "'
                     EventID:string '"'
                     Attributes:string
                     ;
let Changes = Common
| where MessageType == 1
| parse DistinctPart with '"'
                     ADObject:string '" "'
                     EventID:string '" "'
                     EventType:string '" '
                     Attributes:string
                     ;
let Attacks = Common
| where MessageType == 2
| parse DistinctPart with '"'
                     Codename:string '" "'
                     Severity:string '" "'
                     SourceHostname:string '" "'
                     SourceIP:string '" "'
                     DestinationHostname:string '" "'
                     DestinationIP:string '" "'
                     Attributes:string
                     ;
```

Reply:
It used to work, in fact the version on master doesn't have the fix you propose. has anything changed to break the parser?

My internship at Alsid has ended so I do not have access to my Azure subscription anymore, is there any way I could test the parser myself with the sample data?

Reply:
@shainw thanks for the feedback. I've added some examples in the mentioned files, is that better now?

Reply:
@Aorimn - Thanks for adding the additional message type.  I am still seeing the parser not handle the parsing correctly. 
![image](https://user-images.githubusercontent.com/45466083/110171594-cc493d00-7db0-11eb-9486-568affd8b257.png)

It looks to me like the escaping of the double quote is not being handled properly.  When I change to this, it works:

```
| parse RawData with
                     Time:datetime ' '
                     Host:string ' '
                     Product:string '['
                     PID:int ']: \"'
                     MessageType:int '\" \"'
                     AlertID:int '\" \"'
                     Forest:string '\" \"'
                     Domain:string '\" '
                     DistinctPart:string;
let Deviances = Common
| where MessageType == 0 
| parse DistinctPart with '"'
                     Codename:string '" "'
                     Severity:string '" "'
                     ADObject:string '" "'
                     DevianceID:string '" "'
                     ProfileID:string '" "'
                     ReasonCodename:string '" "'
                     EventID:string '"'
                     Attributes:string
                     ;
let Changes = Common
| where MessageType == 1
| parse DistinctPart with '"'
                     ADObject:string '" "'
                     EventID:string '" "'
                     EventType:string '" '
                     Attributes:string
                     ;
let Attacks = Common
| where MessageType == 2
| parse DistinctPart with '"'
                     Codename:string '" "'
                     Severity:string '" "'
                     SourceHostname:string '" "'
                     SourceIP:string '" "'
                     DestinationHostname:string '" "'
                     DestinationIP:string '" "'
                     Attributes:string
                     ;
```